system_intelligence/ram_info.py

```diff
@@ -1,190 +1,261 @@
-                          bank['slot'],
-                          f'{bytes_to_hreadable_string(bank["memory"])} /'
-                          f'{bytes_to_hreadable_string(ram_info["total"])}',
-                          hz_to_hreadable_string(bank['clock']))
-        console.print(table)
-
-        table = create_styled_table('Random-Access Memory Cache')
-
-        table.add_column('Slot', justify='left')
-        table.add_column('Physid', justify='left')
-        table.add_column('Capacity', justify='left')
-
-        for cache in ram_info['cache']:
-            table.add_row(cache['slot'], cache['physid'], bytes_to_hreadable_string(cache['capacity']))
-
-        console.print(table)
+
+class RamInfo(BaseInfo):
+    """
+    Get info on RAM
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.RAM_TOTAL = psutil is not None
+        # the pyudev package is only available for linux, so just import if the current OS is linux
+        if self.OS == 'linux':
+            import pyudev
+            pyudev.Context()
+        else:
+            print('[bold yellow]Unable to import package pyudev. RAM information may be limited.')
+
+    def query_ram(self, sudo: bool = False, **kwargs) -> t.Mapping[str, t.Any]:
+        """
+        Get all available information about RAM.
+        """
+        total_ram = self.query_ram_total()
+        ram: t.Dict[str, t.Any] = {'total': total_ram, 'banks': {}}
+
+        # query ram info for MacOS
+        if self.OS == 'darwin':
+            self.query_ram_macos(ram)
+        elif self.OS == 'win32':
+            self.query_ram_windows(ram)
+        else:
+            ram_banks, ram_cache = self.query_ram_banks_cache(sudo=sudo, **kwargs)
+            ram['cache'] = {}
+            if ram_banks:
+                ram['banks'] = ram_banks
+            if ram_cache:
+                ram['cache'] = ram_cache
+
+        return ram
+
+    def query_ram_total(self) -> t.Optional[int]:
+        """
+        Get information about total available RAM.
+        """
+        if not self.RAM_TOTAL:
+            return None
+
+        return psutil.virtual_memory().total
+
+    def query_ram_macos(self, ram: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+        """
+        Query RAM info on MacOS
+        """
+        cmd = (['system_profiler', 'SPMemoryDataType'])
+        ram_info = subprocess.check_output(cmd).decode('utf-8')
+        # split the BANKs (like "slots") into different items
+        ram_banks = [e.strip() for e in ram_info.split('BANK') if e][1:]
+
+        # for each RAM slot, extract details
+        for i in range(0, len(ram_banks)):
+            ram_slot_details = [e.strip() for e in ram_banks[i].split('\n') if e]
+            ram_slot_details_set = {el.split(':')[0].strip(): el.split(':')[1].strip() for el in ram_slot_details}
+            ram['banks']['BANK ' + ram_slot_details[0]] = {k: v for k, v in ram_slot_details_set.items() if k in
```

Comment:
still no a fan of k, v but we can keep it here :)

Reply:
Will rename, something along `atribute, value`